.github/workflows/deploy.yml

```diff
@@ -0,0 +1,126 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ 'develop', 'master', 'staging' ]
+  pull_request:
+    branches: [ 'develop', 'master' ]
+
+jobs:
+  cancel-previous:
+    name: 'Cancel Previous Runs'
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: cancel-previous
+    env:
+      SB_APP_DIR: /home/travis/switchboard-dapp
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: '10.22.0'
+          registry-url: https://registry.npmjs.org/
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-node-
+            ${{ runner.os }}-
+
+      - name: Install npm packages
+        run: |
+          npm install
+          envsubst < src/environments/constants.template.ts > src/environments/constants.ts
+
+      - name: Run Lint
+        run: |
+          npm run lint
+
+      - name: Run Test
+        run: |
+          npm run test:headless
+
+      - name: Prepare for master branch
+        if: github.ref == 'refs/heads/master'
+        run: |
+          cp src/environments/environment.prod.ts src/environments/environment.ts
+          echo "Prod Env Enabled"
+          npm run build-prod
+
+      - name: Prepare for non-master branch
+        if: github.ref != 'refs/heads/master'
+        run: |
+          cp src/environments/environment.dev.ts src/environments/environment.ts
+          echo "Prod Env Disabled"
+          npm run build
+
+      - name: Deploy on develop branch
+        if: github.ref == 'refs/heads/DOI-887_migrate_switchboard_to_GH'
```

Comment:
i think here should be `refs/heads/develop`

Reply:
ohh, I forgot to remove that.

Reply:
fixed.